webpack/components/SelectAllCheckbox/index.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Dropdown, DropdownToggle, DropdownToggleCheckbox,
+  DropdownItem } from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+
+import './SelectAllCheckbox.scss';
+
+const SelectAllCheckbox = ({
+  selectAll,
+  selectNone,
+  selectPage,
+  selectedCount,
+  pageRowCount,
+  totalCount,
+  areAllRowsOnPageSelected,
+  areAllRowsSelected,
+}) => {
+  const [isSelectAllChecked, setSelectAllChecked] = useState(null);
+  const [isSelectAllDropdownOpen, setSelectAllDropdownOpen] = useState(false);
+
+  // Checkbox states: false = unchecked, null = partially-checked, true = checked
+  // Flow: All are selected -> click -> none are selected
+  // Some are selected -> click -> none are selected
+  // None are selected -> click -> page is selected
+  const onSelectAllCheckboxChange = () => {
+    if (isSelectAllChecked === false) {
+      return selectPage();
+    }
+    return selectNone();
+  };
+  const onSelectAllDropdownToggle = () => setSelectAllDropdownOpen(isOpen => !isOpen);
+
+  const handleSelectAll = () => {
+    setSelectAllDropdownOpen(false);
+    selectAll();
+  };
+  const handleSelectPage = () => {
+    setSelectAllDropdownOpen(false);
+    selectPage();
+  };
+  const handleSelectNone = () => {
+    setSelectAllDropdownOpen(false);
+    selectNone();
+  };
+
+  useEffect(() => {
+    let newCheckedState;
+    if (selectedCount === 0) {
+      newCheckedState = false;
+    } else if (selectedCount > 0) {
+      newCheckedState = null; // null is partially-checked state
+    } else if (areAllRowsSelected) {
+      newCheckedState = true;
+    }
+    setSelectAllChecked(newCheckedState);
+  }, [selectedCount, areAllRowsSelected]);
+
+  const selectAllDropdownItems = [
+    <DropdownItem key="select-none" component="button" onClick={handleSelectNone}>
+      {`${__('Select none')} (0)`}
+    </DropdownItem>,
+    <DropdownItem key="select-page" component="button" isDisabled={areAllRowsOnPageSelected} onClick={handleSelectPage}>
+      {`${__('Select page')} (${pageRowCount})`}
+    </DropdownItem>,
+    <DropdownItem key="select-all" component="button" isDisabled onClick={handleSelectAll}>
+      {`${__('Select all')} (${totalCount})`}
```

Comment:
@MariSvirik , @jeremylenz : Hiding this instead of disabling it for pages that don't support multi page select would allow us to use this on other pages without having to wait for API support..Not sure how soon that would be..

Reply:
@sjha4 Updated.